4-js/index.js

```diff
@@ -0,0 +1,330 @@
+const allDelete = document.getElementById('all-delete');
+const arrUserCheckbox = document.getElementsByClassName('user-checkbox');
+const companyInput = document.getElementById('company-input');
+const nameInput = document.getElementById('name-input');
+const addressInput = document.getElementById('address-input');
+const inputFilter = document.getElementsByClassName('input-filter');
+const titleNewUser = document.getElementById('title-new-user');
+const titleEditUser = document.getElementById('title-edit-user');
+const paginator = document.getElementById('paginator');
+const paginations = document.getElementsByClassName('paginator-number');
+
+document.addEventListener('DOMContentLoaded', () => {
+  findArrLenghtUsers();
+  addInputListener();
+});
+
+addUserBtn.addEventListener('click', () => {
+  const arrFormUserInput = document.getElementsByClassName('form-user-input');
+  formUser.classList.remove('is-hide');
+  createUserBtn.classList.remove('is-hide');
+  titleNewUser.classList.remove('is-hide');
+  createUserBtn.setAttribute('disabled', 'disabled');
+  createUserBtn.classList.add('is-translucent');
+  for (let i = 0; i < arrFormUserInput.length; i++) {
+    arrFormUserInput[i].addEventListener('keyup', formCheck);
+  }
+});
+
+createUserBtn.addEventListener('click', () => {
+  createUser();
+  formUser.classList.add('is-hide');
+  createUserBtn.classList.add('is-hide');
+  titleNewUser.classList.add('is-hide');
+  userCompany.value = '';
+  userName.value = '';
+  userAddress.value = '';
+  userCity.value = '';
+  userCountry.value = '';
+});
+
+allDelete.addEventListener('click', () => {
+  deleteSelectedUsers();
+});
+
+mainCheckbox.addEventListener('click', () => {
+  for (let i = 0; i < arrUserCheckbox.length; i++) {
+    if (mainCheckbox.checked !== arrUserCheckbox[i].checked) {
+      arrUserCheckbox[i].checked = !arrUserCheckbox[i].checked;
+    }
+  }
+  allDelete.classList.toggle('is-hide');
+});
+
+editUserBtn.addEventListener('click', () => {
+  editUser();
+  userCompany.value = '';
+  userName.value = '';
+  userAddress.value = '';
+  userCity.value = '';
+  userCountry.value = '';
+  formUser.classList.add('is-hide');
+  editUserBtn.classList.add('is-hide');
+  titleEditUser.classList.add('is-hide');
+});
+
+function findArrLenghtUsers() {
+  sendRequsest('GET',requestUrl).then((data) => {
+    createPaginator(data.length)
+  })
+  .catch((err) => console.log(err));
+}
+
+function createPaginator(length) {
+  const quantityPagination = Math.ceil(length / 10);
+  for (let i = 1; i <= quantityPagination; i++) {
+    let button = document.createElement('button');
+    button.innerText = i;
+    button.name = i * 10;
+    button.classList.add('paginator-number');
+    paginator.appendChild(button);
+    if (i === 1) {
+      button.classList.add('active');
+    }
+    button.addEventListener('click',showPagination);
+  }
+  getUsersInfo('_start=' + (quantityPagination * 10 - 10) + '&_limit=50');
```

Comment:
I would recommend make all calculations in variable and pass it, it is more readable

Reply:
done